packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -859,19 +916,108 @@ function useEffectImpl(fiberEffectTag, hookEffectTag, create, deps): void {
     }
   }
 
-  currentlyRenderingFiber.effectTag |= fiberEffectTag;
+  sideEffectTag |= fiberEffectTag;
   workInProgressHook.memoizedState = pushEffect(
     hookEffectTag,
     create,
     destroy,
     nextDeps,
   );
+}
+
+function mountEffect(
+  create: () => mixed,
+  deps: Array<mixed> | void | null,
+): void {
   if (__DEV__) {
-    currentHookNameInDev = null;
+    currentHookNameInDev = 'useEffect';
+  }
+  const hook = mountWorkInProgressHook();
```

Comment:
Let's move all of these `mountWorkInProgressHook()` calls into `mountEffectImpl()`. That way this is just one function call indirections instead of two.

Reply:
Alternatively, inline mountEffectImpl since it's so small.